src/shapes/RoundedBox.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { forwardRef, useMemo } from 'react'
+import { Mesh, Shape, ExtrudeBufferGeometry } from 'three'
+import { useUpdate } from 'react-three-fiber'
+
+const eps = 0.00001
+
+function createShape(width: number, height: number, radius0: number) {
+  const shape = new Shape()
+  const radius = radius0 - eps
+  shape.absarc(eps, eps, eps, -Math.PI / 2, -Math.PI, true)
+  shape.absarc(eps, height - radius * 2, eps, Math.PI, Math.PI / 2, true)
+  shape.absarc(width - radius * 2, height - radius * 2, eps, Math.PI / 2, 0, true)
+  shape.absarc(width - radius * 2, eps, eps, 0, -Math.PI / 2, true)
+  return shape
+}
+
+type Props = {
+  width: number
+  height: number
+  depth: number
+  radius0: number
```

Comment:
`{ width = 1, height = 1, depth = 1, radius0 = 0.1, smoothness = 4, children, ...rest },`

I would be in favor of having sane defaults for these dimensions and make them optional.
Then a user could just `<RoundedBox />`  and get a default box that is `1m` cubed.

Also, +1 to `radius0` needing a rename. It does look a bit awkward ;)

Reply:
Done